vertx-mysql-client/src/test/java/io/vertx/mysqlclient/MySQLUnixDomainSocketTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.mysqlclient;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import io.vertx.sqlclient.PoolOptions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.testcontainers.shaded.org.apache.commons.lang.SystemUtils;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assume.assumeTrue;
+
+@RunWith(VertxUnitRunner.class)
+public class MySQLUnixDomainSocketTest {
+
+  private static final String unixSocketFile = System.getProperty("unix.socket.file");
+
+  private MySQLPool client;
+  private MySQLConnectOptions options;
+
+  @Before
+  public void before() {
+    assumeTrue(SystemUtils.IS_OS_UNIX);
```

Comment:
Unix is ambiguous here and we also haven't determined how we test in OS X environment with Docker.

Reply:
Yes, I've updated this check Linux only for now